sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -304,6 +311,11 @@ def run(self, pipeline):
     self.dataflow_client = apiclient.DataflowApplicationClient(
         pipeline._options)
 
+    # Upload the original proto for the pipeline
+    self.dataflow_client.stage_file(self.job.google_cloud_options.staging_location,
```

Comment:
<!--new_thread; commit:d9ce45cc29f40f0d0a656dd0333423d18491e4df; resolved:0-->
Could you do this within stage_job_resources (https://github.com/apache/beam/blob/1974b920e4b3bbe8549e25fe789f9dada13c1769/sdks/python/apache_beam/runners/dataflow/internal/apiclient.py#L478)

Reply:
You want me to plumb the original `pipeline` through to there? It isn't part of the `job`.

Reply:
I am conflicted about what to do here. I do not think we should plumb the pipeline. At the same time I want to keep all staging to happen in a single location. 

You can leave a TODO here to figure out what to do about it. Perhaps we can find a middle ground as a list of additional files/buffers that needs to be staged could be passed to the dataflow_client.